src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java

```diff
@@ -16,15 +17,161 @@ public abstract class SparkMaxMotor implements IMotor {
 	protected final SparkMaxWrapper motor;
 	private final String logPath;
 	private final ConnectedInputAutoLogged connectedInput;
+	private SparkBase.Warnings warnings;
+	private SparkBase.Faults faults;
 
 	public SparkMaxMotor(String logPath, SparkMaxWrapper motor) {
 		this.logPath = logPath;
 		this.motor = motor;
+		this.warnings = motor.getWarnings();
+		this.faults = motor.getFaults();
 
 		this.connectedInput = new ConnectedInputAutoLogged();
 		connectedInput.connected = true;
 
+		createAlerts();
+	}
+
+	public void createAlerts() {
 		AlertManager.addAlert(new PeriodicAlert(Alert.AlertType.ERROR, logPath + "disconnectedAt", () -> !isConnected()));
+		createFaultAlerts();
+		createWarningAlerts();
+	}
+
+	private void createFaultAlerts() {
+		//@formatter:off
+		AlertManager.addAlert(
+			new PeriodicAlert(
+					Alert.AlertType.ERROR,
+					logPath + "OtherErrorAt",
+					() -> faults.other
+			)
+		);
```

Comment:
I would have put this last

Reply:
The alert creations are the same order the faults variables are in the Faults object, I think it's correct to leave it this way